api/specs/web-server/_projects_crud.py

```diff
@@ -84,6 +84,7 @@ async def list_projects(
             example='{"field": "last_change_date", "direction": "desc"}',
         ),
     ] = '{"field": "last_change_date", "direction": "desc"}',
+    filters: Annotated[Json | None, Query()] = None,
```

Comment:
Q: I thought we already had a searching mechanism in there? then not?

Reply:
This is inspired from docker rest API (which maps almost one-to-one with docker's CLI counterpart ) where `search` is out of the definition of `filter` and rather implemented in a separate entrypoint (BTW we have `projects:search`.

`filters` concept in docker api is more as:
1. the filters are encoded in a single query parameter `filters` as a JSON encoded map
2. each filter refer to a specific attribute of the collection's resource  (in this case `projects`)
3. each filter is customized, i.e. basically you can create any filter criterias

SEE for instance these
![image](https://github.com/user-attachments/assets/9412b9ed-2281-489e-a03d-c15e3ec04121)

Reply:
so that means the frontend calls 2 different endpoints? what happens when I search in the frontend? it calls search gets a list of project or list projects and gets another list of projects?
what if I want to search in the trashed projects?

Reply:
@sanderegg Good question. Both the LIST and SEARCH  have equivalent `filters`. Therefore i can filter trashed items in both cases. 

The fact that we have a `search` option in LIST and **also** a SEARCH entrypoint is IMO inconsistent but it was a front-end requirment

Reply:
I like the idea of these filters very much. But my challenge is: How do I know if I successfully managed to apply a filter or not. I.e. are these filters validated against a Pydantic model which will let me know if I had a typo in one of my filters? I have experienced this challenge when using the aws cli where I find it tricky to know if I have successfully applied a filter or not.